src/Windows/Avalonia.Win32.Automation/Marshalling/ComVariant.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Avalonia.Win32.Automation.Marshalling;
+
+#if NET7_0_OR_GREATER
+// Oversimplified ComVariant implementation based on https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ComVariant.cs
+// Available 
+[StructLayout(LayoutKind.Explicit)]
+internal struct ComVariant : IDisposable
+{
+    // VARIANT_BOOL constants.
+    internal const short VARIANT_TRUE = -1;
+    internal const short VARIANT_FALSE = 0;
+
+    // Most of the data types in the Variant are carried in _typeUnion
+    [FieldOffset(0)] private TypeUnion _typeUnion;
+
+    [StructLayout(LayoutKind.Sequential)]
+    private struct TypeUnion
+    {
+        public ushort _vt;
+        public ushort _wReserved1;
+        public ushort _wReserved2;
+        public ushort _wReserved3;
+
+        public UnionTypes _unionTypes;
+    }
+
+    [StructLayout(LayoutKind.Explicit)]
+    private unsafe struct UnionTypes
+    {
+        [FieldOffset(0)] public sbyte _i1;
+        [FieldOffset(0)] public short _i2;
+        [FieldOffset(0)] public int _i4;
+        [FieldOffset(0)] public long _i8;
+        [FieldOffset(0)] public byte _ui1;
+        [FieldOffset(0)] public ushort _ui2;
+        [FieldOffset(0)] public uint _ui4;
+        [FieldOffset(0)] public ulong _ui8;
+        [FieldOffset(0)] public int _int;
+        [FieldOffset(0)] public uint _uint;
+        [FieldOffset(0)] public short _bool;
+        [FieldOffset(0)] public int _error;
+        [FieldOffset(0)] public float _r4;
+        [FieldOffset(0)] public double _r8;
+        [FieldOffset(0)] public long _cy;
+        [FieldOffset(0)] public double _date;
+        [FieldOffset(0)] public IntPtr _bstr;
+        [FieldOffset(0)] public IntPtr _unknown;
+        [FieldOffset(0)] public IntPtr _dispatch;
+        [FieldOffset(0)] public IntPtr _pvarVal;
+        [FieldOffset(0)] public IntPtr _byref;
+        [FieldOffset(0)] public SafeArray*parray;
+        [FieldOffset(0)] public SafeArray**pparray;
+    }
+
+    /// <summary>
+    /// Release resources owned by this <see cref="ComVariant"/> instance.
+    /// </summary>
+    public unsafe void Dispose()
```

Comment:
Won't [VariantClear](https://learn.microsoft.com/en-us/windows/win32/api/oleauto/nf-oleauto-variantclear) be sufficient?

Reply:
I guess it would need a `this` pointer which might be a problem for structs.